File: carpartsshop/src/main/java/fsega/project/carpartsshop/model/Invoice.java
package fsega.project.carpartsshop.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Invoice {
    @EmbeddedId
    private InvoiceKey invoiceKey;

    @MapsId("customerId")
    @ManyToOne(fetch = FetchType.EAGER)
    private Customer customer;

    @MapsId("carPartId")
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private CarPart carPart;

    private String details;
    private Date date;
}

File: carpartsshop/angularclient/src/app/model/invoice/invoice.ts
import {CarPart} from "../carPart/car-part";
import {Customer} from "../customer/customer";

export class Invoice {
  invoiceKey: InvoiceKey | undefined;
  customer: Customer | undefined;
  carPart: CarPart | undefined;
  details: string | undefined;
  date: Date | undefined;


  constructor(invoiceKey: InvoiceKey | undefined, customer: Customer | undefined, carPart: CarPart | undefined, details: string | undefined, date: Date | undefined) {
    this.invoiceKey = invoiceKey;
    this.customer = customer;
    this.carPart = carPart;
    this.details = details;
    this.date = date;
  }
}

export class InvoiceKey{
  carId: Number | undefined;
  customerId: Number | undefined;


  constructor(carId: Number | undefined, customerId: Number | undefined) {
    this.carId = carId;
    this.customerId = customerId;
  }
}
